driver/src/main/java/org/neo4j/driver/internal/Bookmark.java

```diff
@@ -0,0 +1,154 @@
+/*
+ * Copyright (c) 2002-2017 "Neo Technology,"
+ * Network Engine for Objects in Lund AB [http://neotechnology.com]
+ *
+ * This file is part of Neo4j.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.neo4j.driver.internal;
+
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Iterator;
+import java.util.Map;
+
+import org.neo4j.driver.v1.Value;
+
+import static java.util.Collections.emptyMap;
+import static java.util.Collections.singleton;
+import static org.neo4j.driver.v1.Values.value;
+
+public final class Bookmark
+{
+    private static final String BOOKMARK_KEY = "bookmark";
+    private static final String BOOKMARKS_KEY = "bookmarks";
+    private static final String BOOKMARK_PREFIX = "neo4j:bookmark:v1:tx";
+
+    private static final long UNKNOWN_BOOKMARK_VALUE = -1;
+
+    private static final Bookmark EMPTY = new Bookmark( Collections.<String>emptySet() );
+
+    private final Iterable<String> values;
+    private final String maxValue;
+
+    private Bookmark( Iterable<String> values )
+    {
+        this.values = values;
+        this.maxValue = maxBookmark( values );
+    }
+
+    public static Bookmark empty()
+    {
+        return EMPTY;
+    }
+
+    public static Bookmark from( String value )
+    {
+        if ( value == null )
+        {
+            return empty();
+        }
+        return from( singleton( value ) );
+    }
+
+    public static Bookmark from( Iterable<String> values )
+    {
+        if ( values == null )
+        {
+            return empty();
+        }
+        return new Bookmark( values );
+    }
+
+    public boolean isEmpty()
+    {
+        return maxValue == null;
+    }
+
+    public String asString()
```

Comment:
This method name is confusing in a way that I would question why only return one bookmark rather than all of them back (like toString method).
Maybe change to names like asSingleString(), or maxBookmarkAsString, or singleBookmarkAsString?